internal/fs/vss_windows.go

```diff
@@ -0,0 +1,979 @@
+	return vss.convertToVSSAsync(oleIUnknown, err)
+}
+
+// convertToVSSAsync looks up IVSSAsync interface if given result
+// is a success.
+func (vss *IVssBackupComponents) convertToVSSAsync(
+	oleIUnknown *ole.IUnknown, err error) (*IVSSAsync, error) {
+	if err != nil {
+		return nil, err
+	} else {
+		comInterface, err := queryInterface(oleIUnknown, UIID_IVSS_ASYNC)
+		if err != nil {
+			return nil, err
+		}
+
+		iVssAsync := (*IVSSAsync)(unsafe.Pointer(comInterface))
+		return iVssAsync, nil
+	}
+}
+
+// convertGUIDToUintptr converts the Data4 part for VSS api calls
+func convertGUIDToUintptr(guid *ole.GUID) (uintptr, uintptr) {
+	a := uint32(ole.IID_NULL.Data4[0]) | (uint32(ole.IID_NULL.Data4[1]) << 8) |
+		(uint32(ole.IID_NULL.Data4[2]) << 16) | (uint32(ole.IID_NULL.Data4[3]) << 24)
+	b := (uint32(ole.IID_NULL.Data4[4]) << 32) | (uint32(ole.IID_NULL.Data4[5] << 40)) |
+		(uint32(ole.IID_NULL.Data4[6] << 48)) | (uint32(ole.IID_NULL.Data4[7]) << 56)
+	return uintptr(a), uintptr(b)
+}
+
+// IsVolumeSupported calls the equivalent VSS api.
+func (vss *IVssBackupComponents) IsVolumeSupported(volumeName string) (bool, error) {
+	volumeNamePointer, err := syscall.UTF16PtrFromString(volumeName)
+	if err != nil {
+		panic(err)
+	}
+
+	isSupported := false
+	var result uintptr = 0
+
+	if runtime.GOARCH == "386" {
+		ptrToID := uintptr(unsafe.Pointer(&ole.IID_NULL))
+
+		result, _, _ = syscall.Syscall9(vss.getVTable().isVolumeSupported, 6,
+			uintptr(unsafe.Pointer(vss)), uintptr(*(*uintptr)(unsafe.Pointer(uintptr(ptrToID)))),
+			uintptr(*(*uintptr)(unsafe.Pointer(uintptr(ptrToID + 4)))),
+			uintptr(*(*uintptr)(unsafe.Pointer(uintptr(ptrToID + 8)))),
+			uintptr(*(*uintptr)(unsafe.Pointer(uintptr(ptrToID + 12)))),
+			uintptr(unsafe.Pointer(volumeNamePointer)), uintptr(unsafe.Pointer(&isSupported)), 0, 0)
+	} else {
+		result, _, _ = syscall.Syscall6(vss.getVTable().isVolumeSupported, 4,
+			uintptr(unsafe.Pointer(vss)), uintptr(unsafe.Pointer(ole.IID_NULL)),
+			uintptr(unsafe.Pointer(volumeNamePointer)), uintptr(unsafe.Pointer(&isSupported)), 0, 0)
+	}
+
+	return isSupported, newVssErrorIfResultNotOK("IsVolumeSupported() failed", HRESULT(result))
+}
+
+// StartSnapshotSet calls the equivalent VSS api.
+func (vss *IVssBackupComponents) StartSnapshotSet() (ole.GUID, error) {
+	var snapshotSetId ole.GUID
+	result, _, _ := syscall.Syscall(vss.getVTable().startSnapshotSet, 2,
+		uintptr(unsafe.Pointer(vss)), uintptr(unsafe.Pointer(&snapshotSetId)), 0,
+	)
+
+	return snapshotSetId, newVssErrorIfResultNotOK("StartSnapshotSet() failed", HRESULT(result))
+}
+
+// AddToSnapshotSet calls the equivalent VSS api.
+func (vss *IVssBackupComponents) AddToSnapshotSet(volumeName string, idSnapshot *ole.GUID) error {
+	volumeNamePointer, err := syscall.UTF16PtrFromString(volumeName)
+	if err != nil {
+		panic(err)
+	}
+
+	var result uintptr = 0
+
+	if runtime.GOARCH == "386" {
+		a, b := convertGUIDToUintptr(ole.IID_NULL)
```

Comment:
Please also change this to use the GUID passing pattern from `DeleteSnapshots` and then remove `convertGUIDToUintptr`.

Reply:
Fixed in https://github.com/restic/restic/pull/2274/commits/8135b85c6e700ec9640f1644cc176c3a703b00c5